build-support/bin/ci.py

```diff
@@ -175,30 +196,85 @@ def get_remote_execution_oauth_token_path() -> Iterator[str]:
     yield tf.name
 
 # -------------------------------------------------------------------------
-# Blacklists
+# Blacklists and whitelists
 # -------------------------------------------------------------------------
 
-def get_blacklisted_targets(filename: str) -> Set[str]:
-  return {
-    line.strip()
-    for line in Path(f"build-support/ci_blacklists/{filename}").read_text().splitlines()
-  }
+Target = str
+TargetSet = Set[Target]
 
 
-def get_all_python_tests(*, tag: Optional[str] = None) -> Set[str]:
-  command = [
-    "./pants.pex",
-    "--filter-type=python_tests",
-    "filter",
-    "src/python::",
-    "tests/python::",
-    "contrib::"
-  ]
-  if tag is not None:
-    command.insert(1, f"--tag={tag}")
-  return set(subprocess.run(
-    command, stdout=subprocess.PIPE, encoding="utf-8", check=True
-  ).stdout.strip().split("\n"))
+class DefaultTestBehavior(Enum):
```

Comment:
Should this just be `TestBehavior`? The position in which it is used is what defines whether it is a "default" or not.